client/src/googleAuth/signIn.js

```diff
@@ -0,0 +1,46 @@
+import React, { useState, useEffect } from 'react';
+import { notification } from 'antd';
+import { renderButton, checkSignedIn } from './authUtils';
+import DayVisitsReport from '../components/Charts/dayVisitsReport';
+
+function GoogleLogin() {
+  const [isSignedIn, setIsSignedIn] = useState(false);
+
+  const updateSignin = (signedIn) => {
+    setIsSignedIn(signedIn);
+    if (!signedIn) {
+      renderButton();
+    }
+  };
+
+  const init = async () => {
```

Comment:
I think you could wrap this in `useCallback`, and then the function will be created in memory once and then reused instead of being re-created every time this component renders.
https://reactjs.org/docs/hooks-reference.html#usecallback